### PR TITLE
Added hints, updated versions and added js client

### PR DESCRIPTION
**Pull request by Woody4618**
### Problem
While going through our hello world user tests some problems emerged. 

### Summary of Changes
Some of these are tackled in this PR. For example adding a link to the solana web faucet, adding solutions for some pit falls  and updating everything to the newest versions. 

Also added a JS client to the hello world guide to be able to call the program in the end for a nice little success.

### CODE REVIEW

File: content/guides/getstarted/setup-local-development.md [hunk 1]
```diff
@@ -71,28 +72,36 @@ lightweight VM that instantly starts when you need it.
 
 First start with
 [installing WSL](https://learn.microsoft.com/en-us/windows/wsl/install) on your
-system. Be sure to restart your computer when installation is done, then
-continue this guide.
+system. Type the following command in your Windows power shell:
```
Comment (mikemaccana) [inline content/guides/getstarted/setup-local-development.md:75]:
```suggestion
system. Type the following command in Windows PowerShell:
```

File: content/guides/getstarted/local-rust-hello-world.md [hunk 2]
```diff
@@ -210,15 +215,156 @@ Program Id: EFH95fWg49vkFNbAdw9vy75tM7sWZ2hQbTTUmuACGip3
...
+  signature: txHash,
+});
+
+console.log(
+  `Congratulations! Look at your ‘Hello World’ transaction in the Solana Explorer:
+  https://explorer.solana.com/tx/${txHash}?cluster=custom`,
+);
+```
+
+> Don't forget to replace `YOUR_PROGRAM_ID` with the program id you got from the
```
Comment (mikemaccana) [inline content/guides/getstarted/local-rust-hello-world.md:316]:
```suggestion
> Don't forget to replace `YOUR_PROGRAM_ID` with the program ID you got from the
```

**Review by mikemaccana — APPROVED** [review]
A couple of minor suggestions but looks fine.

File: content/guides/getstarted/setup-local-development.md [hunk 3]
```diff
@@ -292,6 +323,17 @@ on your machine:
 solana-test-validator
 ```
 
+> In WSL you may need to first navigate to a folder where you have default write
+> access:
+>
+> ```shell
+> cd home
+> cd <your username>
```
Comment (nickfrosty) [inline content/guides/getstarted/setup-local-development.md:331]: In unix system, the user can nagivate to their home directory using the `~` path:

```suggestion
> cd ~
```

File: content/guides/getstarted/setup-local-development.md [hunk 4]
```diff
@@ -359,7 +401,11 @@ solana airdrop 2
 
 The `solana airdrop` command has a limit on how many SOL tokens can be requested
 _per airdrop_ for each cluster (testnet or devnet). If your airdrop transaction
-fails, lower your airdrop request quantity and try again.
+fails, lower your airdrop request quantity and try again. You can also get your
+wallet address using `solana address` and request testing sol on the
+[solana web faucet](https://faucet.solana.com). If you ever need bigger amounts
+you can find ways to do so in this
+[guide](https://solana.com/de/developers/guides/getstarted/solana-token-airdrop-and-faucets)
```
Comment (nickfrosty) [inline content/guides/getstarted/setup-local-development.md:408]:
```suggestion
also get your
wallet address using `solana address` and request testing SOL on the
[Solana Web Faucet](https://faucet.solana.com). If you ever need bigger amounts
you can find ways to do so in this
[guide](/content/guides/getstarted/solana-token-airdrop-and-faucets.md)
```

File: content/guides/getstarted/local-rust-hello-world.md [hunk 5]
```diff
@@ -20,6 +20,7 @@ keywords:
   - blockchain developer
   - blockchain tutorial
   - web3 developer
+  - windows
```
[the remark this answers is not in the document]
Comment (nickfrosty) [reply content/guides/getstarted/local-rust-hello-world.md:23]: this does not matter for keywords

File: content/guides/getstarted/local-rust-hello-world.md [hunk 6]
```diff
@@ -178,14 +179,18 @@ Inside a terminal window, you can build your Solana Rust program by running in
...
 > the build path of your compiled program's `.so` file and the default keyfile
-> that will be used for the program's address. `cargo build-bpf` installs the
+> that will be used for the program's address. `cargo build-sbf` installs the
 > toolchain from the currently installed solana CLI tools. You may need to
-> upgrade those tools if you encounter any version incompatibilities.
+> upgrade those tools if you encounter any version incompatibilities. In case
+> you get an error like:
+> `error while loading shared libraries: librustc_driver-278a6e01e221f788.so`you
+> may need to go to `~/.cache/solana/` and `rm -rf` the plattform tools there
+> and then run `cargo build-sbf` again.
```
Comment (nickfrosty) [inline content/guides/getstarted/local-rust-hello-world.md:193]: typo fix:
```suggestion
> may need to go to `~/.cache/solana/` and `rm -rf` the platform tools there and
> then run `cargo build-sbf` again.
```

File: content/guides/getstarted/local-rust-hello-world.md [hunk 7]
```diff
@@ -210,15 +215,156 @@ Program Id: EFH95fWg49vkFNbAdw9vy75tM7sWZ2hQbTTUmuACGip3
...
+For MacOS you can follow this
+[guide](https://nodejs.org/en/download/package-manager)
+
+### Create the client file
+
+Install the Solana web3.js library and the Solana helpers library:
+
+```shell
+npm install @solana/web3.js
+npm install @solana-developers/helpers
```
Comment (nickfrosty) [inline content/guides/getstarted/local-rust-hello-world.md:256]: you can install multiple packages using the same line:
```suggestion
npm install @solana/web3.js @solana-developers/helpers
```

File: content/guides/getstarted/local-rust-hello-world.md [hunk 8]
```diff
@@ -210,15 +215,156 @@ Program Id: EFH95fWg49vkFNbAdw9vy75tM7sWZ2hQbTTUmuACGip3
...
+```shell
+Congratulations! Look at your ‘Hello World’ transaction in the Solana Explorer:
+  https://explorer.solana.com/tx/2fTcQ74z4DVi8WRuf2oNZ36z7k9tGRThaRPXBMYgjMUNUbUSKLrP6djpRUZ8msuTXvZHFe3UXi31dfgytG2aJZbv?cluster=custom
+```
+
+<Callout type="success" title="Congrats">
+If you follow the link you should be able to see your 'Hello World' transaction on the Solana explorer.
+</Callout>
+
+## Deploy to solana devnet
```
Comment (nickfrosty) [inline content/guides/getstarted/local-rust-hello-world.md:338]:
```suggestion
## Deploy to Solana devnet
```

File: content/guides/getstarted/local-rust-hello-world.md [hunk 9]
```diff
@@ -210,15 +215,156 @@ Program Id: EFH95fWg49vkFNbAdw9vy75tM7sWZ2hQbTTUmuACGip3
...
+Now you have successfully deployed your program to your local cluster. If you
+want to deploy it to the public devnet to show your program to your friends you
+can do so by running the following command:
+
+```shell
+solana program deploy ./target/deploy/hello_world.so --url https://api.devnet.solana.com
+```
+
+and then change the connections url in your `client.mjs` also to
+`https://api.devnet.solana.com` and run the client again.
```
Comment (nickfrosty) [inline content/guides/getstarted/local-rust-hello-world.md:349]:
```suggestion
Then change the connections url in your `client.mjs` also to
`https://api.devnet.solana.com` and run the client again.
```

**Review by nickfrosty — CHANGES_REQUESTED** [review]
Nice additions, just some small typo fixes

**Review by nickfrosty — APPROVED** [review]
Thanks for these updates Jonas :)

**Pull request merged**